pkg/serviceaccount/util.go

```diff
@@ -25,11 +25,35 @@ import (
 
 // UserInfo returns a user.Info interface for the given namespace, service account name and UID
 func UserInfo(namespace, name, uid string) user.Info {
-	return &user.DefaultInfo{
-		Name:   apiserverserviceaccount.MakeUsername(namespace, name),
-		UID:    uid,
-		Groups: apiserverserviceaccount.MakeGroupNames(namespace),
+	return (&ServiceAccountInfo{
+		Name:      name,
+		Namespace: namespace,
+		UID:       uid,
+	}).UserInfo()
+}
+
+type ServiceAccountInfo struct {
+	Name, Namespace, UID string
+	PodName, PodUID      string
+}
+
+func (sa *ServiceAccountInfo) UserInfo() user.Info {
+	const (
+		podNameKey = "kubernetes.io/pod-name"
```

Comment:
I looked around for prior art for a label like this, and found:
- For [flexvolumes](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/flexvolume/driver-call.go#L55), we use `kubernetes.io/pod.name`
- For [statefulsets](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/apps/v1/types.go#L31), we use `statefulset.kubernetes.io/pod-name`.

I think the names you've got are fine, but we should try to be somewhat standard w/ stuff like this going forward.

Reply:
Switched it to `kubernetes.io/pod.name` and `kubernetes.io/pod.uid`